src/App.tsx

```diff
@@ -15,36 +17,159 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const buttonsFromServer = [
+  'Sort alphabetically',
+  'Sort by length',
+  'Reverse',
+  'Reset',
+];
+
+enum SortField {
+  Name = 'name',
+  Length = 'length',
+}
+
+// const SORT_FIELD_NAME = 'name';
+// const SORT_FIELD_LENGTH = 'length';
+
+interface ButtonProps {
+  button: string;
+  sortField: string;
+  setSortField: React.Dispatch<React.SetStateAction<string>>;
+  reverse: boolean;
+  setReverse: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function getPreparedButtons(
+  buttons: string[],
+  { sortField, reverse }: { sortField: string; reverse: boolean },
+) {
+  const preparedButtons = [...buttons];
+
+  if (!sortField && !reverse) {
+    preparedButtons.pop();
+  }
+
+  return preparedButtons;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: { sortField: string; reverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SortField.Name) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField === SortField.Length) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+const Button: React.FC<ButtonProps> = ({
+  button,
+  sortField,
+  reverse,
+  setSortField,
+  setReverse,
+}) => {
   return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+    <button
+      type="button"
+      className={cn(
+        'button',
+        { 'is-info': button === 'Sort alphabetically' },
+        { 'is-success': button === 'Sort by length' },
+        { 'is-warning': button === 'Reverse' },
+        { 'is-danger': button === 'Reset' },
+        {
+          'is-light': button === 'Sort alphabetically' && sortField !== 'name',
+        },
+        { 'is-light': button === 'Sort by length' && sortField !== 'length' },
+        { 'is-light': button === 'Reverse' && !reverse },
+        { 'is-light': button === 'Reset' },
+      )}
+      onClick={() => {
+        const value = button;
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+        switch (value) {
+          case 'Sort alphabetically':
+            setSortField(SortField.Name);
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+            break;
+          case 'Sort by length':
+            setSortField(SortField.Length);
```

Comment:
The task requires using `enum SortField` instead of string literals for sorting options. Please refactor this code to use `SortField.Name` and `SortField.Length` instead of the string literals 'Sort alphabetically' and 'Sort by length'.